test/integration/httpInfrastructureRest.spec.ts

```diff
@@ -0,0 +1,642 @@
+      const result = await client.httpClientFailure.head401();
+      assert.equal(result.status, "401");
+    });
+
+    it("head410 should return 410", async () => {
+      const result = await client.httpClientFailure.head410();
+      assert.equal(result.status, "410");
+    });
+
+    it("delete407 should return 407", async () => {
+      try {
+        const result = await client.httpClientFailure.delete407();
+      } catch (error) {
+        if (isNode) {
+          assert.equal(error.statusCode, 407);
+        } else {
+          assert.equal(error.code, "REQUEST_SEND_ERROR");
+        }
+      }
+    });
+
+    it("head429 should return 429", async () => {
+      const result = await client.httpClientFailure.head429();
+      assert.equal(result.status, "429");
+    });
+
+    // TODO: Investigate why options calls are sending back 204
+    it.skip("options400 should return 400", async () => {
+      const result = await client.httpClientFailure.options400();
+      assert.equal(result.status, "400");
+    });
+
+    it.skip("options403 should return 403", async () => {
+      const result = await client.httpClientFailure.options403();
+      assert.equal(result.status, "403");
+    });
+
+    it.skip("options412 should return 412", async () => {
+      const result = await client.httpClientFailure.options412();
+      assert.equal(result.status, "412");
+    });
+
+    it("patch400 should return 400", async () => {
+      const result = await client.httpClientFailure.patch400();
+      assert.equal(result.status, "400");
+    });
+
+    it("patch405 should return 405", async () => {
+      const result = await client.httpClientFailure.patch405();
+      assert.equal(result.status, "405");
+    });
+
+    it("post400 should return 400", async () => {
+      const result = await client.httpClientFailure.post400();
+      assert.equal(result.status, "400");
+    });
+
+    it("post406 should return 406", async () => {
+      const result = await client.httpClientFailure.post406();
+      assert.equal(result.status, "406");
+    });
+
+    it("post415 should return 415", async () => {
+      const result = await client.httpClientFailure.post415();
+      assert.equal(result.status, "415");
+    });
+
+    it("put400 should return 400", async () => {
+      const result = await client.httpClientFailure.put400();
+      assert.equal(result.status, "400");
+    });
+
+    it("put404 should return 404", async () => {
+      const result = await client.httpClientFailure.put404();
+      assert.equal(result.status, "404");
+    });
+
+    it("put409 should return 409", async () => {
+      const result = await client.httpClientFailure.put409();
+      assert.equal(result.status, "409");
+    });
+
+    it("put413 should return 413", async () => {
+      const result = await client.httpClientFailure.put413();
+      assert.equal(result.status, "413");
+    });
+  });
+
+  describe("Failure scenarios", () => {
```

Comment:
I think these are really "redirect" scenarios right? Can you update the title please?

Reply:
changed